### ISSUE TITLE
loss값 계산 관련

### ISSUE THREAD

**Issue by kalibianw** [Justin-A/DeepLearning101#16]
안녕하세요, loss값 계산 부분에 관련하여 issue 남기게 되었습니다.

p.136의 코드를 예시로 들면 
```
def evaluate(model, test_loader):
    model.eval()
    test_loss = 0
    correct = 0
    with torch.no_grad():
        for image, label in test_loader:
            image = image.to(DEVICE)
            label = label.to(DEVICE)
            output = model(image)
            test_loss += criterion(output, label).item()
            prediction = output.max(1, keepdim=True)[1]
            correct += prediction.eq(label.view_as(prediction)).sum().item()
    test_loss /= len(test_loader.dataset)
    test_accuracy = 100. * correct / len(test_loader.dataset)
    
    return test_accuracy, test_loss
```
이 부분에서 `test_loss` 변수에 `criterion(output, label).item()`의 값을 모두 저장한 후 `len(test_loader.dataset)`로 나눠주는 부분이 있는데, 이 부분을 `test_loss /= (len(test_loader.dataset) / BATCH_SIZE)`로 바꾸면 loss값이 Epoch 한 번의 평균 값으로 출력이 잘 됩니다.

`criterion = nn.CrossEntropyLoss()`를 수행할 때 `reduction` argument의 기본 값이 `"mean"`이므로 해당 Batch에 Loss의 평균 값이 리턴되는 것으로 이해했는데(https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html 를 참고했습니다.), 특별히 다른 값을 지정하지 않고 사용하였으니 `test_loss /= (len(test_loader.dataset) / BATCH_SIZE)` 이처럼 나눠주는 것이 맞다고 생각합니다.

이 부분 어떻게 생각하시나요??

_Originally posted by @kalibianw in https://github.com/Justin-A/DeepLearning101/issues/14#issuecomment-803822748_

**Comment by Justin-A** [comment]
안녕하세요, Justin입니다.
관련된 내용을 확인하여 답변드립니다.

for 반복문을 통해 계산된 test_loss는 각 배치에 해당되는 loss값이며, 10,000개의 test_set에 대해서 313개의 배치가 존재하므로,
이에 대한 평균값이 반환되기 때문에, 지적해주신 test_loss /= (len(test_loader.dataset) / BATCH_SIZE) 로 수정되어야 된다고 생각됩니다.

관련된 내용은 다른 코드에도 적용되어 수정하도록 하겠습니다.

지적해주셔서 감사합니다.
